Remove memory addresses from anonymous class frames

toemat opened the pull request:
Match the new `${addr}` syntax as well as the previous `0x{addr}` syntax when stripping out the memory address so that events can group correctly again.

For https://github.com/getsentry/sentry-php/issues/1314

stayallive commented:
Could you also add a test so we don't regress in the future?

toemat commented:
Added a new test and updated CHANGELOG, thank you for the guidance! 🙌

ste93cry commented on tests/StacktraceTest.php:
```diff
@@ -241,6 +241,45 @@ public function buildFromBacktraceDataProvider(): \Generator
             ],
         ];
+
+        yield 'Backtrace with frame containing memory address $ format' => [
```
What about changing the description to something more meaningful, e.g. `Backtrace with frame containing memory address in PHP >=7.4.2 format`?

ste93cry commented on tests/StacktraceTest.php:
```diff
@@ -203,7 +203,7 @@ public function buildFromBacktraceDataProvider(): \Generator
         ];
 
-        yield 'Backtrace with frame containing memory address' => [
+        yield 'Backtrace with frame containing memory address 0x format' => [
```
What about changing the description to something more meaningful, e.g. `Backtrace with frame containing memory address in PHP <7.4.2 format`?

ste93cry reviewed (changes requested):
Just some small changes and nitpicks, but good work overall

ste93cry commented:
I just realized that you are targeting the wrong branch. Since this is a bugfix, it should be merged into `master`

toemat commented:
Thank you @ste93cry, I really appreciate the review and guidance. I addressed the comments (should I respond to them, or 'resolve' them?). I'll switch the target to `master`.

ste93cry commented:
>should I respond to them, or 'resolve' them?

I don't think there is a right answer. Usually, I mark them as done if there's nothing more to add, and I leave a comment otherwise.

>I addressed the comments

Thank you for your work! A release will follow soon 🚢
